assignment4.Rmd

```diff
@@ -107,12 +118,35 @@ sweep(loadings, 2, colSums(loadings), "/") #sweep() computes each row as a propo
 
 #You can generate a biplot to help you, though these can be a bit confusing. They plot the transformed data by the first two components. Therefore, the axes represent the direction of maximum variance. Then mapped onto this point cloud are the original directions of the variables, depicted as red arrows. It is supposed to provide a visualization of which variables "go together". Variables that possibly represent the same underlying construct point in the same direction.  
 
-biplot(pca)
+biplot(pca, cex = 0.5, arrow.len = 0.01)
 
 #Calculate values for each student that represent these your composite variables and then create a new correlogram showing their relationship to mean_correct.
 
+D5 <- as.matrix(D2) %*% pca$rotation
+
 ```
 # Part III  
 
 ## Also in this repository is a data set collected from class that shows how many students thought that a TC program was related to andother TC program. Use PCA to look for components of related programs. Explain why you think there are relationships between these programs.
 
+PC1 is more about policy, politics, leadership and PC2 is more about health, neuroscience,physiology. I feel that the variances number make similar programs closely relate to each other.
+
+```{r}
+
+TC1 <- read.csv("TCprograms.csv", header = TRUE, sep = " ")
+TC1 <- subset(TC1, select = -program)
+COR3 <- cor(TC1)
+corrplot(COR3, order="AOE", method="circle", tl.pos="lt", type="upper",        
+tl.col="black", tl.cex=0.6, tl.srt=45, 
+        addCoef.col="black", addCoefasPercent = TRUE,
+        sig.level=0.50, insig = "blank")
+pca1 <- prcomp(TC1,scale. = TRUE, center = TRUE)
+summary(pca1)
+plot(pca1,type = "lines")
+pca1$rotation
+loadings1 <- abs(pca1$rotation)
+sweep(loadings, 2, colSums(loadings1), "/")
+biplot(pca1,cex = 0.2, arrow.len = 0.01)
```

Comment:
Nice work! You can interpret PCs by looking at eigenvectors (rotation) and pay attention to those variables with a coefficient larger than 0.2 (absolute value).